subsys/bluetooth/host/hci_core.c

```diff
@@ -261,7 +261,7 @@ int bt_hci_cmd_send_sync(u16_t opcode, struct net_buf *buf,
 	cmd(buf)->sync = &sync_sem;
 
 	/* Make sure the buffer stays around until the command completes */
-	net_buf_ref(buf);
+	/* net_buf_ref(buf); */
```

Comment:
what about the case where the buffer is provided by the caller to `bt_hci_cmd_send_sync()`? you don't know what refcount is in that buf in that case.

Reply:
When bt_hci_cmd_send_sync() returns an error, the buffer will not be released, I think this is incorrect.

Reply:
Yeah, maybe you should use other methods to fix this problem.